meteor/sofie-server/business-logic/services/rundown-event-builder-implementation.ts

```diff
@@ -10,6 +10,15 @@ import { AdLibPiece } from '../../model/entities/ad-lib-piece'
 import { Piece } from '../../model/entities/piece'
 
 export class RundownEventBuilderImplementation implements RundownEventBuilder {
+	public buildDeletedEvent(rundown: Rundown): RundownEvent {
+		return {
+			type: RundownEventType.DELETED,
+			rundownId: rundown.id,
+			segmentId: '',
+			partId: '',
```

Comment:
They are required by the `RundownEvent` return type.
Considering that a few of the other event builder methods have the same issue, perhabs those should be marked optional.
Getting @LindvedKrvang's input on this might be nice.

Reply:
I am torn on this one myself.

It would be nice to have a type for Events that all events could use which makes listening for them easier, but at the same time, it doesn't make sense to have attributes that are never used.

One solution would be to make a type for each event we have i.e. `RundownActivatedEvent`, `RundownTakeEvent` and so forth, but a lot of these events would be identical save for one attribute that's most likely an enum. That seems like way overkill.

I am up for suggestions.